torch/csrc/jit/tensorexpr/stmt.h

```diff
@@ -417,6 +434,7 @@ class For : public StmtNode<For> {
   For(const Var* var, const Expr* start, const Expr* stop, Stmt* body)
       : var_(var), start_(start), stop_(stop) {
     CHECK(var && start && stop && body);
+    body = clone_if_has_parent(body);
```

Comment:
Instead of a silent clone. Why is this not an error, and the caller has to explicitly make a clone itself? 

If the caller didn't expect this to happen, the CHECK is very valuable in chasing down what is happening.

Reply:
Good point, I updated the PR!